### DISCUSSION TITLE
what will happen in emergency landing when fail throttle set to 1000 with airmode enabled?

### DISCUSSION THREAD

**Discussion by lida2003** [iNavFlight/inav#9181] [Q&A]
I have a few questions about airmode feature.

a)How airmode works? 

b)does emergency landing get airmode feature?

c)so what will happen in emergency landonh when fail throttle set to 1000 with airmode enabled?

ps: failsafe throttle 1000/hover throttle 1500, default settings.

**Comment by Jetrell** [comment]
We touched on this, on Discord last week.

A) For multicopters,  Airmode is designed to keep the PID loop active and allow attitude stabilization when the throttle is lowered, in freestyle flight maneuvers.. To accomplish this, it basically only requires the _integral_ part of the PID loop.  

B) Only if its permanently enabled or enabled with FS in the modes tab.

C) Air mode will try to maintain the quads present attitude as its descending. 
But in your case (a tumble) it will not succeeded, because the integral term by its function, can not respond fast enough to the error. It would require the P-term to act first, to halt the spin.

Hover throttle is only applied when there is trusted sensor data.

**Reply by lida2003** [reply]
> Hover throttle is only applied when there is trusted sensor data.

it's uable or trusted sensor data when baro data is ok and gps is invalid?

**Reply by Jetrell** [reply]
Yes, but that can only be known by looking at the log.

**Reply by lida2003** [reply]
Now, it's close to what the scenario I wanna talk about.

There are two options here for emergency landing:
a) `hover_throttle` when `>=EST_USABLE`
b) `failsafe_throttle` when `<EST_USABLE`

And worst case is `failsafe_throttle`, if we have this case activated `failsafe_throttle = 1000`, is air mode able to maintain present attitude?

If we can't maintain present attitude, what't difference between emergency lading and `FAILSAFE_PROCEDURE_DROP_IT`?

PS: that's what I have mentioned in [tumble (prop stall???)](https://github.com/iNavFlight/inav/discussions/9167) when emergency lading.

**Reply by breadoven** [reply]
Emergency Landing doesn't disarm ... FAILSAFE DROP_IT does disarm immediately IF Failsafe is active .

At worst an Emergency Landing without Failsafe active will use Idle Throttle if Failsafe is set to DROP. This is essentially a drop but with attitude control, at least so long as airmode is working. What is your setting for `airmode_type` ? Also I don't suppose you can remember where the throttle stick was when the Emergency Landing started by any chance, i.e was it at minimum ?

**Reply by lida2003** [reply]
> At worst an Emergency Landing without Failsafe active will use Idle Throttle if Failsafe is set to DROP. This is essentially a drop but with attitude control, at least so long as airmode is working. What is your setting for `airmode_type `?

`set airmode_type = THROTTLE_THRESHOLD`, maybe the backup dump file will give much more details. 
 [INAV_6.1.1_cli_20230627_122419_dump.txt](https://github.com/iNavFlight/inav/files/12093374/INAV_6.1.1_cli_20230627_122419_dump.txt)

> Also I don't suppose you can remember where the throttle stick was when the Emergency Landing started by any chance, i.e was it at minimum ?

Emergency landing happened when RTL. I haven't touch the throttle stick until the last moment, And I think it's around ~46%.

**Reply by breadoven** [reply]
OK ... in that case it's not clear why the quad tumbled given airmode should have been active and maintained attitude control even at 0 throttle.

**Reply by lida2003** [reply]
Yeah, it seems like a puzzle without the log. 

BTW, The sound of props, when RTL swithed to emergency landing, it sounded like propeller stall. So I think maybe as @Jetrell said `failsafe_throttle` = 1150 for conservative default land settings.

If this physical phenomenon (propeller stall at  `failsafe_throttle` = 1000) is possible, then the default parameters  `failsafe_throttle` need to be adjusted.

**Comment by lida2003** [comment]
Just add a link to [ Emergency landing tumbled crash? -- inav 6.1.1 #9184 ](https://github.com/iNavFlight/inav/issues/9184)
